### ISSUE TITLE
"Annotated" cannot be instantiated

### ISSUE THREAD

**Issue by jnooree** [microsoft/pyright#8956]
**Describe the bug**

Instantiating `typing.Annotated` reports a call issue (`reportCallIssue`).

**Code or Screenshots**
```python
# pyright: reportCallIssue=true
from typing import Annotated

Test = Annotated[int, "Test annotation"]
test = Test()
```

**VS Code extension or command-line**

I'm using pyright via VSCode + Pylance.

- VSCode version: `1.93.0`
- Python extension version: `v2024.14.0`
- Pylance extension version: `v2024.9.1`

**Comment by erictraut** [comment]
This behavior is consistent with the [Python typing specification](https://typing.readthedocs.io/en/latest/spec/qualifiers.html#id4), which indicates that `Annotated` is not instantiable and should be flagged by type checkers.

Here's the specific wording in the spec:

> An attempt to call Annotated (whether parameterized or not) should be treated as a type error by type checkers.
